### ISSUE TITLE
Secure Boot support and testing for QEMU platform

### ISSUE THREAD

**Issue by macpijan** [Dasharo/dasharo-issues#538]
**Dasharo version (if applicable)**

v0.1.0

**Dasharo variant (if applicable)**

Emulation QEMU Q35

**Affected component(s) or functionality (if applicable)**

edk2/OvmfPkg and/or https://github.com/Dasharo/open-source-firmware-validation

**Brief summary**

- [ ] Adjust QEMU start scripts so OS (Ubuntu 22.04) installation is possible nad prove that by performing installation - the more automated the process is, the better
   - consider using [automated installer](https://github.com/Dasharo/preseeds#ubuntu) or netboot.xyz, or other
- [ ]  [Secure Boot Suite](https://github.com/Dasharo/open-source-firmware-validation/blob/main/dasharo-security/secure-boot.robot) shall pass under QEMU

**Additional context**

**Comment by pietrushnic** [comment]
@macpijan why do we not rely on LBT002.001?

**Comment by pietrushnic** [comment]
Shouldn't we consider DTS as a testing vehicle for UEFI Secure Boot? I guess it aligns a lot with our other activities.

@TomaszAIR @artur-rs cc

**Comment by macpijan** [comment]
@pietrushnic I prepared some branch if you want to move forward with that: https://github.com/Dasharo/open-source-firmware-validation/pull/71/files

At first, the SB cannot even be enabled manually. This is the first work item, perhaps to be done on the edk2 side.

**Comment by macpijan** [comment]
The problem with using PiKVM for hosting USB images is also relevant for other boards: https://github.com/Dasharo/open-source-firmware-validation/issues/68

**Comment by pietrushnic** [comment]
> At first, the SB cannot even be enabled manually. This is the first work item, perhaps to be done on the edk2 side.

To make sure we integrate on top of the new UX made by @miczyg1 ?

**Comment by macpijan** [comment]
@pietrushnic New UX is another thing. But the problem will be the same most likely. Just try to enable SB manually first to make sure it is feasible.

**Comment by pietrushnic** [comment]
@macpijan, good point. I will try that first and let you know if it works correctly.

**Comment by pietrushnic** [comment]
@miczyg1 @macpijan as it was mentioned on matrix yesterday, the problem we hit we being unable to change `Attempt Secure Boot` button to visible is:

```shell
error: GetVariable ("dbDefault): Not Found
Cannot enroll db: Not Found
```

I'm pretty sure OVMF would not allow for such an issue because OVMF is used massively in the cloud environment. So, most likely, something is wrong with our implementation. I can check the most recent code of OVMF to see if that changes anything.

But that raises the question to which I got an answer from @miczyg1 "Pointeless", namely, will rebase to the most recent EDK II give us an advantage in regards to builds for supported hardware?
